### PR TITLE
jazz up Dwarves similarly to Merfolk

### PR DESCRIPTION
**Pull request by RojjaCebolla**
vanilla races look a little dull next to Splicehack's new races.
this is a simple attempt to make dwarves a smidge more dwarfy.

### CODE REVIEW

**Comment by RojjaCebolla** [comment]
Also for your consideration: DNH puts racial skills into u_init.c instead of putting them into weapon.c
https://github.com/Chris-plus-alphanumericgibberish/dNAO/blob/compat-3.19.1/src/u_init.c#L2449

And Fourk has a more ambitious racial skill boost function; instead of _setting_ skill to a value, it _boosts_ the skill by 1 starting from the base level of the player's role:
(you could squash these together) https://github.com/tsadok/nhfourk/commit/d3a10f0b75989e7be83bcfc006bdbea7769944c6 https://github.com/tsadok/nhfourk/commit/df537de6ec1a7c4f0f6d6b58023499eda7afeefe

**Comment by NullCGT** [comment]
Are dwarves supposed to start with booze? It looks like you added the booze struct, but it isn't used anywhere.

**Comment by RojjaCebolla** [comment]
> Are dwarves supposed to start with booze? It looks like you added the booze struct, but it isn't used anywhere.

whoops, thanks!  used fixup and force pushed

File: src/u_init.c [hunk 1]
```diff
@@ -1107,6 +1110,7 @@ u_init()
         knows_object(DWARVISH_RING_MAIL);
         knows_object(DWARVISH_CLOAK);
         knows_object(DWARVISH_ROUNDSHIELD);
+	knows_object(POT_BOOZE);
```
Comment (NullCGT) [inline src/u_init.c:1113]: Knows_object(POT_BOOZE) is actually unnecessary. As long as dwarves start with a potion of booze, then potions of booze will be type-identified by ini_inv().

**Comment by NullCGT** [comment]
Once the knows_object(POT_BOOZE) is addressed, I'm happy to merge this in. I love the flavor of dwarves always having booze. It feels very "Dwarf Fortress" to me.

**Comment by RojjaCebolla** [comment]
Done!  I would have liked to also make this change: https://github.com/FredrIQ/fiqhack/commit/af5e7538b126ea721c0fb30301011dd0f9856b91 but i wasn't able to "translate" that into NH3-speak yet.  

And that's a not-entirely-dwarf-specific change anyways, it's also potentially of interest to earlygame Arc and Tou who might use a pick until they find a better weapon.

**Comment by RojjaCebolla** [comment]
Well, I messed this up again;  4ff87ac doesn't work, and i think it's because dwarf monsters are set up before the program gets to the "initinv" https://github.com/RojjaCebolla/SpliceHack/blob/4ff87ac3ff34a631f5cc359b35f08d127fd0b2b0/src/makemon.c#L889 part of the file.  I can just revert this commit, or let me know if you have a suggested fix!

**Comment by RojjaCebolla** [comment]
i'll make a new PR for monsterdwarfs carrying booze; for now, this PR is cleaned up (again!)

**Comment by copperwater** [comment]
Going to note the same implications of letting all dwarvish characters get to Expert in pick-axe:
* Arcs and Convicts already had this cap.
* Bar and Val gets a minor boost from Skilled to Expert; given their propensity for melee weapons this isn't a problem.
* Cartomancers go from nothing to Expert; highly significant as they previously had only three Expert weapon options and none were a particularly great melee choice.
* Knights go from Basic to Expert, which is significant as their only expert weapons before were long sword and lance. 
* Monks go from nothing to Expert; highly significant since Monks don't get above-Basic skill in _any_ weapons.
* Priests go from nothing to Expert; significant since they normally only get Expert skill in blunt weapons.
* Rangers go from nothing to Expert; notable since they normally only get Expert in ranged weapons (and short sword).
* Wizards go from nothing to Expert; highly significant since they normally only get Expert in a few weapons.

The booze thing is great, though.

**Comment by RojjaCebolla** [comment]
well, ideally, i'd use a nifty "boost by 1 or 2 levels" function like the one in Fourk, instead of simply _setting_ it to expert.  Not that confident with C++ coding yet.

But anyway, given that Grayswandir (or Gae Bulg, X-Brand) is pretty much always the best idea regardless of what character you're playing or what skills you have access to... i see no problem with different races having a flavorful boost to lesser-used weapon skills!  and for the elves, this can be a compelling wish alternative to Master Key of Thiefy! 😁

**Comment by NullCGT** [comment]
I've implemented this, but without the skill changes for now.

**Pull request closed without merging**